pythonscript/_godot_script.pxi

```diff
@@ -166,10 +168,25 @@ cdef api godot_pluginscript_script_manifest pythonscript_script_init(
     # TODO: possible bug if res:// is not part of PYTHONPATH
     # Remove `res://`, `.py` and replace / by .
     modname = path[6:].rsplit(".", 1)[0].replace("/", ".")
+    
+    is_reload = modname in sys.modules
+    if is_reload:
+        # if we are in a reload, remove the class from loaded classes,
+        # and call importlib.reload to reload the code
+        cls = get_exposed_class(modname)
+
+        # don't try to reload modules that are not managed by godot - e.g. libs or other modules - user should take care of that
+        if cls:
+            if get_pythonscript_verbose():
+                print(f"Reloading python script from {path}")
+            destroy_module(cls=cls)
+            importlib.reload(sys.modules[modname])
+
     try:
-        __import__(modname)  # Force lazy loading of the module
+        importlib.import_module(modname)  # Force lazy loading of the module
```

Comment:
Do we need to do that in the case we were reloading the cls (hence ` importlib.reload(sys.modules[modname])` has just been executed) ?

I guess do this after the reload does nothing (it may slower the code but it's no big deal given reloading a classe shouldn't be considered as a hot path), so it just make the code simpler to let it this way.
If it's the case it may be worth it to add a small comment about this.

Reply:
Yep, if we just reloaded this code does nothing (python loader sees that module is already in sys.modules and just returns what's already loaded). I didn't add an IF statement to make the code simpler. I'll add a comment!